src/main/resources/templates/snowflake/shared/create-stage-qa-task.ssp

```diff
@@ -0,0 +1,38 @@
+<%--
+Copyright 2018 phData Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+--%>
+#import(io.phdata.streamliner.configuration.Snowflake)
+#import(io.phdata.streamliner.configuration.SnowflakeQAOptions)
+<%@ val configuration: io.phdata.streamliner.configuration.Configuration %>
+<%@ val table: io.phdata.streamliner.configuration.SnowflakeTable %>
+#{
+    val destination = configuration.destination.asInstanceOf[Snowflake]
+    val qaOptions = destination.quality.asInstanceOf[SnowflakeQAOptions]
+}#
+CREATE TASK IF NOT EXISTS ${destination.stagingDatabase.name}.${destination.stagingDatabase.schema}.${table.destinationName}_qa_task
+    WAREHOUSE = ${destination.warehouse}
+    SCHEDULE = '${destination.quality.taskSchedule.getOrElse("5 minutes")}'
+    WHEN SYSTEM$STREAM_HAS_DATA('${table.destinationName}_QA_STREAM')
+    AS
+    call table_run_qa('${destination.stagingDatabase.schema}',
+                '${table.destinationName}',
+                ${destination.quality.minimumPercentage},
+                ${destination.quality.minimumCount},
+                ${destination.quality.minimumRuns},
+                ${destination.quality.standardDeviations});
+
+ALTER TASK ${destination.stagingDatabase.name}.${destination.stagingDatabase.schema}.${table.destinationName}_qa_task RESUME;
+
+ALTER TASK ${destination.stagingDatabase.name}.${destination.stagingDatabase.schema}.${table.destinationName}_qa_task SUSPEND;
```

Comment:
How come we spend the the task at the end? Maybe a comment explaining this...

Reply:
I could probably split those out into different templates.